packages/mobile/src/store/products/actions.ts

```diff
@@ -0,0 +1,16 @@
+import { createAsyncThunk } from '@reduxjs/toolkit';
+import { AsyncThunkConfig } from '~/common/types/types';
+import { ProductDto } from '@vse-bude/shared';
+import { ActionType } from './common';
+
+const loadAllProducts = createAsyncThunk<
+  ProductDto[],
+  undefined,
+  AsyncThunkConfig
+>(ActionType.PRODUCTS_FETCH, async (_, { extra }) => {
+  const { productApi } = extra;
+
+  return await productApi.getAllProducts();
```

Comment:
You may skip `await` when returning from `async` function as it still returns a `Promise`
```suggestion
  return productApi.getAllProducts();
```